lldb/tools/lldb-dap/DAP.cpp

```diff
@@ -380,12 +380,19 @@ llvm::json::Value DAP::CreateTopLevelScopes() {
   return llvm::json::Value(std::move(scopes));
 }
 
-ExpressionContext DAP::DetectExpressionContext(lldb::SBFrame &frame,
-                                               std::string &text) {
+static std::string FirstTerm(llvm::StringRef input) {
+  if (input.empty())
+    return "";
+  const auto terms = llvm::getToken(input, " \t\n\v\f\r.[-(");
```

Comment:
if the first term is followed by `.[-(`, couldn't we assume that it's code and not a command?

Reply:
I actually removed the extra characters and left it as whitespace. There are lldb commands like `p/t` or `x/4xw` that can include non alpha numeric characters and users can add aliases, so I think using `ResolveCommand` is the most correct thing to do here.